zanata-war/src/main/java/org/zanata/service/impl/CopyTransWorkFactory.java

```diff
@@ -57,13 +82,374 @@
     @In
     private VersionStateCache versionStateCacheImpl;
 
-    public CopyTransWork createCopyTransWork(HLocale targetLocale,
+    public Integer runCopyTransInNewTx(HLocale targetLocale,
+            HCopyTransOptions options, HDocument document,
+            boolean requireTranslationReview, List<HTextFlow> copyTargets)
+            throws Exception {
+        return runInTransaction(() -> runCopyTrans(targetLocale, options,
```

Comment:
So this is a Runnable/Callable calling an instance method of the outer class? I am not quite sure how the injection work in such condition. The Runnable instance is bound to a stateless scoped bean instance? We also don't make use of the container thread pool, do we? Looks like a ```@Async``` may be better?